### ISSUE TITLE
False positive when using specialization feature and not implementing all items in trait

### ISSUE THREAD

**Issue by supersurviveur** [rust-lang/rust-analyzer#21378]
The following code triggers an error in rust-analyzer while it compiles without any errors or warnings using rust nightly.

**rust-analyzer version**: `rust-analyzer version: 0.3.2735-standalone`

**rustc version**: `rustc 1.94.0-nightly (a6525d526 2025-12-21)`

**editor or extension**: Helix, also tested on Vscode

**code snippet to reproduce**:
```rust
#![allow(incomplete_features)]
#![feature(specialization)]

pub trait Foo {
    fn foo();
}

impl<T> Foo for T {
    default fn foo() {}
}
impl Foo for bool {}
```
**error from rust-analyzer**:
```
not all trait items implemented, missing: `fn foo` (rust-analyzer [E0046])
```

**Comment by ShoyuVanilla** [comment]
I think we should do some heuristic such as:

1. Short circuit if the crate doesn't has the `#![feature(specialization)] enabled
2. If there exists any blanket impl block that has the missing assoc function as a default impl, do not trigger the diagnostic, regardless of whether the impl in problem is a specialization of the blanket impl

instead of accurately checking whether the relevant specialization impl has the default implementation, which is too costly.
